omitnull/omitnull.go

```diff
@@ -492,9 +492,7 @@ func (v *Val[T]) UnmarshalBinary(b []byte) error {
 // sql.Scanner then it will call that.
 func (v *Val[T]) Scan(value any) error {
 	if value == nil {
-		var zero T
-		v.value = zero
-		v.state = StateNull
+		v.value, v.state = *new(T), StateNull
```

Comment:
This seems worse than what was here previously because it allocates.

If I have `Val[int]` then I don't want someone to do `new(int)` to place it on the heap before assigning it to `value`. Can you explain this change further?